packages/headless/src/auth/saml-provider.ts

```diff
@@ -0,0 +1,112 @@
+import {fetch} from 'cross-fetch';
+import {getIsomorphicHistory, IsomorphicHistory} from './isomorphic-history';
+import {getIsomorphicLocation, IsomorphicLocation} from './isomorphic-location';
+
+export interface SamlProviderOptions {
+  organizationId: string;
+  provider: string;
+  request?: Fetch;
+  location?: IsomorphicLocation;
+  history?: IsomorphicHistory;
+}
+
+export interface SamlProvider {
+  exchangeHandshakeToken(): Promise<string>;
+  handshakeTokenAvailable: boolean;
+  login(): void;
+}
+
+type Fetch = (
+  input: RequestInfo,
+  init?: RequestInit | undefined
+) => Promise<Response>;
+
+const handshakeTokenParamName = 'handshake_token';
+
+export function buildSamlProvider(config: SamlProviderOptions): SamlProvider {
```

Comment:
I'm not 100% convinced by this name, because the saml provider might be mixed with the actual provider (e.g. Okta). Open to name suggestions. Some alternatives:

- SamlHelper
- SamlAssistant
- SamlUtility

Reply:
How about `SamlFlow` ?  Since it's about handling the "flow" of doing the login with your browser (redirection), and saving the cookie.

Also, this seems to be an internal module only, right ? No users would interact directly with this.

Reply:
`SamlFlow` is a good suggestion. I will adjust to it. The module is internal, not exposed.